neural_compressor/common/base_tune.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+from typing import Any, Callable, Dict, List, Optional, Union
+
+from neural_compressor.common.base_config import BaseConfig, ComposableConfig
+from neural_compressor.common.logger import Logger
+
+logger = Logger().get_logger()
+
+
+class AlgorithmWrapper:
+    """Abstract base class for wrap algorithm implementation.
+
+    AlgorithmWrapper provides a uniform interface for encapsulating algorithm implementation
+    across different frameworks.
+    This class is intended to be used by a `tuner` to obtain quantized models.
+    """
+
+    def __init__(self, model) -> None:
+        self.model = model
+
+    @abstractmethod
+    def apply(self):
+        """The entry to apply algorithms on a given model."""
+        raise NotImplementedError
+
+
+class TuningObjective:
```

Comment:
shall we explicitly distinguish single objective and multiple objectives? if not, I would suggest to rename as TuningObjectives

Reply:
We don't explicitly distinguish between single and multiple objectives. Renamed it to `TuningObjectives`.